java-backend/src/main/java/org/kframework/backend/java/symbolic/SymbolicRewriter.java

```diff
@@ -740,10 +737,28 @@ private K disjunctResults(List<K> results) {
             global.javaExecutionOptions.log = originalLog;
         }
 
+        List<ConstrainedTerm> tweakedProofResults = proofResults;
+        if (global.javaExecutionOptions.formatFailures && !proofResults.isEmpty()) {
+            for (ConstrainedTerm term : proofResults) {
+                printTermAndConstraint(term);
+            }
+            tweakedProofResults = ImmutableList.of(new ConstrainedTerm(BoolToken.FALSE, initialTerm.termContext()));
+        }
+
         if (global.globalOptions.verbose) {
             printSummaryBox(rule, proofResults, successPaths, step);
         }
-        return proofResults;
+        return tweakedProofResults;
```

Comment:
I don't like that `tweakedProofResults` is returned instead of the actual `proofResults`. Can you do the tweaking as a post-processing step at the time you print them out instead of here?

Reply:
I don't think so. Custom printing is specific to java backend. All I want is to disable default result printing and use custom printing instead, when this option is given. Also I don't want the post-processing that is done on proof results - I want to display final states fully, one by one.

Other commits use this custom output format in different ways, like displaying success final states. So this customization is one way or another inevitable.